connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -544,6 +555,95 @@ public int generation() {
         return generation;
     }
 
+    @Override
+    public synchronized void getConfigDef(final String connType, final Callback<ConfigDefInfo> callback) {
+        log.trace("Submitting get connector configuration definitions request {}", connType);
+
+        addRequest(
```

Comment:
`addRequest` is used to get the processing of this into the main processing loop for group membership. Nothing in here immediately pops out at me as requiring that -- as far as I can tell everything in here is just temporary data that doesn't access any shared state (and I don't think it should need to unless we validate things like conflicting connector names). Can we either remove the `addRequest` or explain why we need it?